### PR TITLE
fix: Workaround for `<afile>` expanding incorrectly

### LINKED ISSUE #369: `TabClosed` callbacks break if `expand("<afile>")` is called prior
Is this an expected behavior or a bug? Because after exploring a little deeper, I found that closing multiple panels is done via `TabClosed` event callbacks, defined here

https://github.com/sindrets/diffview.nvim/blob/86bf6182e2ea767c21711de8a3c396d9b635d970/lua/diffview/init.lua#L44-L50

However, the 'TabClosed' event is specified to be non-recursive (event cannot trigger itself) (I didn't find this in the `:h tabclosed`, but I found this logic in the neovim source code [static void do_autocmd_winclosed(win_T *win)](https://github.com/neovim/neovim/blob/fdc8e966a9183c08f2afec0817d03b7417a883b3/src/nvim/window.c#L3018C45-L3030)), so the end result is that the TabClosed event will not be triggered again when the close operation is performed a second time and the remaining panels will not be closed further down.

Now, I got around this by forcing the event to fire again by calling vim.api.nvim_exec_autocmds in the TabClosed callback, but I'm not sure if that's the right solution, or if the result (only two panels closed) is what you expect.

A small addition, if this is a bug, can you try to let me fix it? I really like this plugin and would like to give some modest contribution.

### PR DESCRIPTION
**Pull request by NGPONG**
Resolves #369.

I have left some question here:

1. Added a comment not related to this pr, sorry if this makes you uncomfortable I will restore it.

2. I'm not sure if the fetch_tabnr's *name* is correct.

3. I'm not sure if fetch_tabnr is in the right place for this, I tried to throw it into *utils* but its logic is not that common.

### CODE REVIEW

File: lua/diffview/init.lua [hunk 1]
```diff
@@ -23,6 +23,22 @@ end
 
 function M.init()
   local au = api.nvim_create_autocmd
+
+  -- Fix the strange behavior that "<afile>" expands non-files
+  -- as file name in some cases.
+  --
+  -- About more information please checkout:
```
Comment (sindrets) [inline lua/diffview/init.lua:30]:
```suggestion
  -- Ref:
```

File: lua/diffview/init.lua [hunk 2]
```diff
@@ -23,6 +23,22 @@ end
...
+  --  * sindrets/diffview.nvim#369
+  --  * neovim/neovim#23943
+  local fetch_tabnr = function (state)
```
Comment (sindrets) [inline lua/diffview/init.lua:33]: Let's just call it `get_tabnr`.

Nit pick: I prefer using Lua's syntactic sugar for defining functions. It's a little denser and it makes a clearer distinction between functions and other types.

```suggestion
  local function get_tabnr(state)
```

**Review by sindrets — CHANGES_REQUESTED** [review]

File: lua/diffview/init.lua (hunk 1 above)
Comment (NGPONG) [reply lua/diffview/init.lua]: fixed

File: lua/diffview/init.lua (hunk 2 above)
Comment (NGPONG) [reply lua/diffview/init.lua]: Great shout! This does get better and has been fixed.

**Review by sindrets — APPROVED** [review]
LGTM, thank you!

**Pull request merged**